### DISCUSSION TITLE
Exposed Keys

### DISCUSSION THREAD

**Discussion by mitraecp** [nuxt/nuxt#16458] [Questions] [answered]
How can i hide this keys?

I thought they would be at least a little hidden. but returning in the exposed document is complicated

![image](https://user-images.githubusercontent.com/75841640/198411289-af11d6fd-7ac8-4ca6-add3-927f6638af20.png)

what am I doing wrong?

**Comment by leccyril** [comment]
Hello, 
on my side i created a server api, then client is calling this server and on this server i use private KEY, here you put it on public and use it on client side => not good

**Comment by TheAlexLichter** [comment, marked as answer]
Everything you use on the frontend is human readable and can be read. As you don't use SSR, you have no real chance to "hide" the keys via `runtimeConfig`.  The best option would be adding an API endpoint (e.g. via Nitro or standalone) and use the keys in there only.

**Reply by leccyril** [reply]
yes it is was i made and explained him, but i have one issue with this for me, first client is calling API , then server api is calling external API with my token, problem i need the result status of the second call as it is a non content POST (return 200 or 400 or 409 or 500), but on client side i received the status of the api call, how can forward directly the result status ? or just modifiy a query with adding header (middleware is not working i saw for this case) ?

**Reply by TheAlexLichter** [reply]
You could just forward the status in your server API call the so it will be preserved ☺️

**Reply by leccyril** [reply]
it is what i've done but i found it not really clean, or maybe it is not the best way to do it ? thanks !
   
    const {status} = await $fetch.raw(REST_API, {
		method: "POST",
		body: recipientWebDTO,
		headers: {
			Authorization: TOKEN,
		},
	}).then((response) => ({
		status: response.status,
	}))
		.catch((error) => ({
			status: error?.response?.status || 500,
		}));
	event.res.statusCode = status;
	return "";

**Reply by leccyril** [reply]
does nuxt3 will use like gateway forward calls or something like that ? interceptor to add some headers etc? it could be great like srping boot does ?

thank you very much !

**Reply by TheAlexLichter** [reply]
You can e.g. try out h3's [`sendProxy`](https://www.jsdocs.io/package/h3#sendProxy). Other than that, [`ohmyfetch`](https://github.com/unjs/ohmyfetch) (`$fetch`) also supports interceptors and more.

**Reply by leccyril** [reply]
hi, how work interceptor ? because i have to add header only in server part cause oh token i have to add

thanks

**Reply by TheAlexLichter** [reply]
See https://github.com/unjs/ohmyfetch#%EF%B8%8F-interceptors ☺️ Also works for `useFetch` ☺️
